Домашняя работа

IsaevMilan opened the pull request:
Сделал новый проект что бы корректно запушить его и сделать Pull requests.

IsaevMilan committed:
init commit

ilshat-abdulin commented:
> Сделал новый проект что бы корректно запушить его и сделать Pull requests.

Привет, Милан. 
При коммите потерялся файл dimen с размерами)

![image](https://user-images.githubusercontent.com/72722086/233775451-633b68bd-f043-4823-a0df-4dd39cfedb82.png)

ilshat-abdulin commented on app/src/main/java/com/example/myplaylistmaker/SettingsActivity.kt:
```diff
@@ -0,0 +1,19 @@
+
+        val backButt = findViewById<Button>(R.id.backButton)
+
+        backButt.setOnClickListener { val searchIntent = Intent (this, MainActivity::class.java )
+            startActivity(searchIntent)
```
🍏 Лучше используй вместо интента, метод finish(), который завершит и скроет текущую активити и перенаправит в предыдущий экран. А через интент создается новый экран

ilshat-abdulin commented on app/src/main/res/layout-v23/activity_main.xml:
```diff
@@ -0,0 +1,83 @@
+    android:background="@color/BackBlue"
+    android:orientation="vertical"
+    android:padding="16dp"
+    android:theme="@style/Theme.MaterialComponents.DayNight.NoActionBar">
```
🍏 Можно удалить. Эта тема и так применяется, т.к. присвоена в AndroidManifest

IsaevMilan replied:
ок)

ilshat-abdulin commented on app/src/main/res/layout-v23/activity_settings.xml:
```diff
@@ -0,0 +1,139 @@
+    android:theme="@style/Theme.MaterialComponents.DayNight.NoActionBar"
+    android:layout_width="match_parent"
+    android:layout_height="match_parent"
+    android:background="#F5F5F5"
```
⚠ Цвет в ресурсы

IsaevMilan replied:
Понял, Уберу все цвета )

ilshat-abdulin commented on app/src/main/res/layout-v23/activity_settings.xml:
```diff
@@ -0,0 +1,139 @@
+        android:layout_width="match_parent"
+        android:layout_height="wrap_content">
+
+        <Button
+
+            android:id="@+id/backButton"
+            android:layout_width="48dp"
+            android:layout_height="48dp"
+            android:layout_marginStart="0dp"
+            android:foreground="?attr/selectableItemBackground"
+            android:backgroundTint="#F5F5F5"
+            app:icon="@drawable/arrow_back"
+            app:iconTint="#30001f"
+            tools:ignore="InvalidId"
+            tools:targetApi="m" />
```
🍏 Необязательно кнопку использовать. Можно ImageView и не придется настраивать backgroundTint, iconTint
С кнопкой виден бэкграунд
![image](https://user-images.githubusercontent.com/72722086/233776166-d2bab254-d7e7-4fad-82e9-286eee7cf72a.png)

IsaevMilan replied:
Попробую изменить мне показалось с кнопкой работать легче))

ilshat-abdulin replied:
Лучше изменить. Видно же как выделяется. По дизайну должно быть все в одной плоскости

ilshat-abdulin commented on app/src/main/res/layout-v23/activity_settings.xml:
```diff
@@ -0,0 +1,139 @@
+            android:layout_marginTop="10dp"
+            android:text="@string/Setting"
+            android:textAllCaps="false"
+            android:textColor="#1A1B22"
```
⚠ В ресурсы

IsaevMilan replied:
ок

ilshat-abdulin commented on app/src/main/res/layout-v23/activity_settings.xml:
```diff
@@ -0,0 +1,139 @@
+    android:layout_height="match_parent"
+    android:background="#F5F5F5"
+    android:orientation="vertical"
+    android:padding="4dp"
```
По моему, можно и без этого отступа обойтись

IsaevMilan replied:
посмотрю как без него.

ilshat-abdulin replied:
Да точно можно) 
Потому что, проверяя отступы компонентов в этом контейнере, постоянно приходится помнить, что у тебя есть эти 4 dp)

ilshat-abdulin commented on app/src/main/res/layout-v23/activity_settings.xml:
```diff
@@ -0,0 +1,139 @@
+            android:minHeight="48dp"
+            android:text="@string/blackth"
+            android:textAllCaps="false"
+            android:textColor="#000000"
```
⚠ В общем, все цвета в ресурсы))

IsaevMilan replied:
я понял))

ilshat-abdulin commented on app/src/main/res/layout-v23/activity_settings.xml:
```diff
@@ -0,0 +1,139 @@
+            android:textSize="@dimen/textSizePlaylist" />
+    </FrameLayout>
+
+    <FrameLayout
+        android:layout_width="match_parent"
+        android:layout_height="61dp"
+        android:layout_marginTop="40dp">
+
+        <Switch
+            android:layout_width="match_parent"
+            android:layout_height="wrap_content"
```
🍏 Можно Switch добавить без FrameLayout - лишняя вложенность только получается.

IsaevMilan replied:
с FrameLayout очень удобно, по крайней мере когда ты новичек))

ilshat-abdulin replied:
Ну как удобно. Мое дело посоветовать)

ilshat-abdulin commented on app/src/main/res/layout-v23/activity_settings.xml:
```diff
@@ -0,0 +1,139 @@
+            android:textSize="16sp"/>
+
+        <ImageView
+            android:layout_width="19dp"
+            android:layout_height="18dp"
```
🍏 Чтобы не присваивать ширине и высоте размеры в каждом ImageView, во-первых, лучше экспортировать из фигмы корректный слой иконки
![image](https://user-images.githubusercontent.com/72722086/233776916-3f0d3c61-f616-4b63-9d2f-66a458df3120.png)

А во-вторых, после корректного экспорта, для иконок ImageView не будет необходимости хардкодить в макете размеры, а достаточно будет присвоить wrap_content

IsaevMilan replied:
спасибо за ценные подсказки.

ilshat-abdulin reviewed (commented):
Привет, Милан. 
При коммите потерялся файл dimen с размерами) А также не хватает файлов разметки activity_media, activity_message, activity_search. Проект не собирается

![image](https://user-images.githubusercontent.com/72722086/233775451-633b68bd-f043-4823-a0df-4dd39cfedb82.png)

IsaevMilan committed:
commit 2
Добавил все файлы которые почему-то пропали в прошлый раз.
Исправил все обязательные замечания, а не обязательные исправил частично так как боюсь испортить)))

IsaevMilan committed:
commit 2
Добавил все файлы которые почему-то пропали в прошлый раз.
Исправил все обязательные замечания, а не обязательные исправил частично так как боюсь испортить)))

IsaevMilan committed:
commit 2
Добавил все файлы которые почему-то пропали в прошлый раз.
Исправил все обязательные замечания, а не обязательные исправил частично так как боюсь испортить)))

ilshat-abdulin commented on app/src/main/res/layout-v23/activity_main.xml:
```diff
@@ -0,0 +1,82 @@
+        android:layout_height="match_parent"
+        android:layout_marginTop="24dp"
+        android:layout_weight="1"
+        android:radius="16dp"
```
Уже используется атрибут cornerRadius, а этот не будет иметь эффекта. И размер повторяется - имеет смысл вынести в ресурсы

ilshat-abdulin commented on app/src/main/res/layout-v23/activity_settings.xml:
```diff
@@ -0,0 +1,138 @@
+
+    <FrameLayout
+        android:layout_width="match_parent"
+        android:layout_height="61dp">
```
🍏 Ты можешь присвоить здесь wrap_content, т.е. размер высоты может определяться за счет компонентов внутри FrameLayout. Например, если посмотреть по дизайну, то верхний и нижний отступ от текста внутри контейнера равен 21dp
![image](https://user-images.githubusercontent.com/72722086/233858116-2fbd2dbf-90e9-440d-9a8e-b4d2a9ac7d00.png)

Можно применить marginTop и marginBottom для TextView. Или проще - marginVertical

```
        <TextView
            android:layout_width="300dp"
            android:layout_height="wrap_content"
            android:layout_marginStart="12dp"
            android:layout_marginVertical="21dp"
            android:layout_gravity="center_vertical"
            android:backgroundTint="@color/settingsWhite"
            android:gravity="start"
            android:text="@string/SHARE"
            android:textAllCaps="false"
            android:textColor="@color/settingsTextBlack"
            android:textSize="@dimen/settingTextSize"/>
```

ilshat-abdulin commented on app/src/main/res/layout-v23/activity_settings.xml:
```diff
@@ -0,0 +1,138 @@
+        <ImageView
+            android:layout_width="wrap_content"
+            android:layout_height="wrap_content"
+            android:layout_marginTop="20dp"
```
🍏 Корректнее будет, если уберешь этот отступ. Вместо это позиционировать в конце по центру, используя end|center для layout_gravity

` android:layout_gravity="end|center"`

ilshat-abdulin commented on app/src/main/res/layout-v23/activity_settings.xml:
```diff
@@ -0,0 +1,138 @@
+            android:layout_width="wrap_content"
+            android:layout_height="wrap_content"
+            android:layout_marginTop="20dp"
+            android:layout_marginEnd="15dp"
```
Молодец, что добавил в проект иконки корректных размеров. Только исправь отступ здесь и в остальных иконках. По дизайну должно быть 12dp от края экрана
![image](https://user-images.githubusercontent.com/72722086/233858450-62ff50b7-580f-40a0-987c-2cd59a93b71d.png)

ilshat-abdulin commented on app/src/main/res/layout-v23/activity_settings.xml:
```diff
@@ -0,0 +1,138 @@
+            android:layout_width="match_parent"
+            android:layout_height="wrap_content"
+            android:layout_gravity="center_vertical"
+            android:layout_marginStart="12dp"
```
Повторяющиеся размеры лучше выноси в ресурсы
